### PR TITLE
refactor PushOptions.GetComponentContext method to a utils func

### LINKED ISSUE #4340: Move "GetComponentContext" to a util package
* In `pkg/odo/cli/url`, we are calling this method which is in `pkg/odo/cli/component`. 
* `URLCreateOptions` and `PushOptions` are used interchangeably (as I can see, `GetComponentContext` is in the method set of `PushOptions` struct).
* Also, this function gets the value which can be used in other CLI packages as well.

It seems like this function belongs to some kind of util package, or other top level file for some component instead of being in `pkg/odo/cli/component/push.go`.

Besides that, should it be removed from the method set of `PushOptions` and put somewhere else? Some kind of common options that are common among various different options viz. `PushOptions`, `URLCreateOptions`, `ComponentOptions`, etc.

_Originally posted by @dharmit in https://github.com/openshift/odo/pull/4311#discussion_r549645173_

/kind discussion
/kind code-refactoring
/good-first-issue

### PR DESCRIPTION
**Pull request by niconosenzo**
**What type of PR is this?**

 /kind code-refactoring

**What does does this PR do / why we need it**:
Moved `GetComponentContext()` from a `PushOptions` method:
 
https://github.com/openshift/odo/blob/88fb37203d0c4782bef09d37187f883327967e89/pkg/odo/cli/component/push.go#L83-L86

To the `cli.utils` package so it can be used in other CLI packages as well:

```go
// GetComponentContext gets the component context
func GetComponentContext(po *component.PushOptions) string {
	return po.ComponentContext
}
```

**Which issue(s) this PR fixes**:

Fixes #4340 

**PR acceptance criteria**:

- [x] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/odo/cli/component/common_push.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type CommonPushOptions struct {
 
 	sourceType       config.SrcType
 	sourcePath       string
-	componentContext string
+	ComponentContext string
```
Comment (dharmit) [inline pkg/odo/cli/component/common_push.go:34]: @mik-dass are we comfortable making this an exported element of the struct instead of private?

**Review by dharmit — COMMENTED** [review]
Looks good to me overall. Requested Mrinal's feedback.

**Comment by dharmit** [comment]
@niconosenzo there are a bunch of unit test failures. Can you PTAL?
```
 pkg/odo/cli/component/common_push.go:56:58: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:65:59: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:74:44: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:96:119: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:121:84: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:194:8: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:195:6: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/common_push.go:224:33: cpo.componentContext undefined (type *CommonPushOptions has no field or method componentContext, but does have ComponentContext)
pkg/odo/cli/component/create.go:361:17: co.PushOptions.componentContext undefined (type *PushOptions has no field or method componentContext)
pkg/odo/cli/component/devfile.go:97:41: po.componentContext undefined (type *PushOptions has no field or method componentContext)
pkg/odo/cli/component/devfile.go:97:41: too many errors
make[1]: *** [vet] Error 2
make[1]: Leaving directory `/go/src/github.com/openshift/odo'
make: *** [openshiftci-presubmit-unittests] Error 2 
```

File: pkg/odo/cli/component/common_push.go (hunk 1 above)
Comment (mik-dass) [reply pkg/odo/cli/component/common_push.go:34]: This will allow the other packages to modify the variable.

Comment (dharmit) [reply pkg/odo/cli/component/common_push.go:34]: I understand that. And that's why I was wondering if we are comfortable with it.

Comment (mik-dass) [reply pkg/odo/cli/component/common_push.go:34]: I don't think we should go ahead with this approach.

Comment (dharmit) [reply pkg/odo/cli/component/common_push.go:34]: @niconosenzo there's a suggestion on what approach might be helpful on the issue https://github.com/openshift/odo/issues/4340#issuecomment-775921042. Do you think you want to continue working on this? 

It's fine if you'd not feel comfortable doing so. There are plenty of other issue on the repo to pick from.

**Comment by mik-dass** [comment]
/hold 
due to the discussion here https://github.com/openshift/odo/pull/4403#pullrequestreview-582237151

**Comment by dharmit** [comment]
Closing this discussion based on discussion with @niconosenzo & @mik-dass on Slack.

/close

**Pull request closed without merging**